### ISSUE TITLE
Group chats fall out of online sync

### ISSUE THREAD

**Issue by rodneyrod** [majn/telegram-purple#330]
I've noticed that after fixing the previous crash bug, bitlbee will still stop syncing with the most up to date conversations shortly after connecting.
The client will attempt to fetch previous conversation and you might get a few new messages synced up, but after a couple of minutes new messages stop coming through, both PM and group chats.

The &bitlbee room still shows users going on and offline (+v or -v) but chats will not update. It will also be usually accompanied by one or more of the group chat channels netsplitting with other users, even though they seem to later rejoin.

**Comment by dequis** [comment]
Sounds similar to #269

**Comment by rodneyrod** [comment]
@dequis Yeah they do look similar, is there anything I could do to help e.g. provide debug logs or something else?

**Comment by rodneyrod** [comment]
Also, if you repeatedly disconnect from Telegram and reconnecting after the chats have fallen out of sync, it will stay in sync with the network for longer and will also sync chat history that was missed from the previous connection.

**Comment by rodneyrod** [comment]
Got another thing with this bug, when it's in this 'outside of sync' state, bitlbee can still send messages to other users in 1-on-1 chats which are shown on other Telegram clients but not to any type of groupchat, and no incoming message is shown on bitlbee.

**Comment by rodneyrod** [comment]
Well, since I've been having issues compiling and getting debugging working with bitlbee myself, I tested it out with Pidgin on another similarly spec'd machine I have around

https://gist.github.com/rodneyrod/a7d7fdb91e8dcb0e1404bc47ea9a11c9

**Comment by rodneyrod** [comment]
Here's the gdb logs from bitlbee, no debug flags though.

https://gist.github.com/rodneyrod/4d2f21ffc9ff7431315ddd3e268ad320

**Comment by rodneyrod** [comment]
Here's a second log, this one captures what happens normally when I normally try to login

https://gist.github.com/rodneyrod/bdff59ae6ac653fdbffb8138165cb4de

**Comment by dequis** [comment]
Those gdb backtraces are useless. They are the result of pressing ctrl-c, interrupting the program in the middle of doing nothing, and getting a backtrace of that. This bug isn't a crash bug so it doesn't make sense to try to use gdb with it.

On the other hand the pidgin log does have a crash ("Pidgin 2.11.0 has segfaulted and attempted to dump a core file [...] please read the instructions at http://developer.pidgin.im/wiki/GetABacktrace") but there's no backtrace for that one. Next time open a new ticket for crash issues, it's not related to this ticket.

**Comment by rodneyrod** [comment]
Thanks for the feedback, can you suggest any ways that I could get some meaningful data out of bitlbee and back to you?

**Comment by dequis** [comment]
I don't know, this is in the telegram-purple bug territory. The console output is as meaningful as it gets. The telegram-purple devs may know more.

**Comment by rodneyrod** [comment]
I've managed to work around this issue by removing some of the larger supergroups I'm subscribed to.
I removed 2 groups that regularly receive 4000+ posts a day each, and afterwards I have been receiving messages from Telegram and groups without issue.

**Comment by rodneyrod** [comment]
So I've managed to trace back the issue to telegram-purple fetching missing posts since last login, or the previous backlog of messages which can only be set to a minimum of 1 day.

If there was a setting where you could disable backlog fetching, I have a strong feeling this would resolve the issue.

**Comment by rodneyrod** [comment]
@dequis I managed to get bitlbee to crash again recently, my terminal emulator over here has cut off the top part of the logs but you have the SIGSEGV at the end, if you need more of the log I can try it again when I'm at home next.

https://gist.github.com/rodneyrod/ff05c639468d53034075576b95f2c9fd

**Comment by BenWiederhake** [comment]
```
Program received signal SIGSEGV, Segmentation fault.
0x00007fffeb121e3b in tglq_query_error (TLS=0x555555863800, id=6380499946907825152) at queries.c:317
317         vlogprintf (E_WARNING, "error for query '%s' #%" INT64_PRINTF_MODIFIER "d: #%d :%.*s\n", q->methods->name, id, error_code, error_len, error);
#0  0x00007fffeb121e3b in tglq_query_error (TLS=0x555555863800, id=6380499946907825152) at queries.c:317
#1  0x00007fffeb10ddd2 in work_rpc_result (c=0xd558616a0, msg_id=<optimized out>, TLS=0x555555863800) at mtproto-client.c:846
#2  rpc_execute_answer (TLS=TLS@entry=0x555555863800, c=0xd558616a0, c@entry=0x5555558616a0, msg_id=<optimized out>) at mtproto-client.c:949
#3  0x00007fffeb10e9fa in process_rpc_message (enc=0x7fffec40cb20 <Response.11256>, len=104, c=0x5555558616a0, TLS=0x555555863800) at mtproto-client.c:1135
```

Oh no, that means tgl crashed while handling an error, so even after fixing this you'll see some errors >_<  (but at least not crashed).

Presumably `q` is `NULL`, because the only other source for SIGSEGV would be reading from `methods->name`, which only happens inside `tglq_query_error`, which doesn't appear on the stacktrace.

Sure, I could probably just monkeypatch this and add a check for `q==NULL` in that function.  But that feels wrong, as I don't think that's supposed to ever happen.  @majn: Ping.

**Comment by majn** [comment]
@BenWiederhake the tgl error handling appears to be complete bogus, just look at this:

  ```if (!q) {
vlogprintf (E_WARNING, "error for query '%s' #%" INT64_PRINTF_MODIFIER "d: #%d :%.*s\n", q->methods->name, id, error_code, error_len, error);
vlogprintf (E_WARNING, "No such query\n");```

So basically if q is NULL then dereference q. This should be easily ficable by removing the method name.

**Comment by BenWiederhake** [comment]
Should be fixed in 1.3.1.  Feel free to reopen if there's any issue.
